src/complete.cpp

```diff
@@ -838,19 +836,24 @@ static bool short_ok(const wcstring &arg, const complete_entry_opt_t *entry,
     return result;
 }
 
+struct cmd_data_t {
+    const wcstring &name;
+    bool has_path;
+};
+
 /// Load command-specific completions for the specified command.
-static void complete_load(const wcstring &name, bool reload) {
-    // We have to load this as a function, since it may define a --wraps or signature.
-    // See issue #2466.
-    function_load(name);
-    completion_autoloader.load(name, reload);
+static void complete_load(const struct cmd_data_t *cmd, bool reload) {
```

Comment:
Since it looks like cmd is never intended to be NULL, I suggest writing this as `const cmd_data_t &cmd` (also note the struct can be omitted if you prefer)